pkg/fetch/http_fetcher_test.go

```diff
@@ -220,6 +221,49 @@ func TestHTTPFetcherFetchBlob(t *testing.T) {
 		require.True(t, proto.Equal(response.BlobDigest, helloDigest.GetProto()))
 		require.Equal(t, response.Status.Code, int32(codes.OK))
 	})
+
+	t.Run("WithAuthHeaders", func(t *testing.T) {
+		request := &remoteasset.FetchBlobRequest{
+			InstanceName: "",
+			Uris:         []string{uri, "www.another.com"},
+			Qualifiers: []*remoteasset.Qualifier{
+				{
+					Name:  "http_header:Authorization",
+					Value: `Bearer anothertoken`,
+				},
+				{
+					Name:  "http_header:Accept",
+					Value: "application/vnd.docker.distribution.manifest.list.v2+json",
+				},
+				{
+					Name:  "http_header_url:0:Authorization",
+					Value: `Bearer letmein1`,
+				},
+				{
+					Name:  "checksum.sri",
+					Value: "sha256-GF+NsyJx/iX1Yab8k4suJkMG7DBO2lGAB9F2SCY4GWk=",
+				},
+			},
+		}
+		matcher := &headerMatcher{
+			headers: map[string]string{
+				"Authorization": "Bearer letmein1",
+				"Accept":        "application/vnd.docker.distribution.manifest.list.v2+json",
+			},
+		}
+		httpDoCall := roundTripper.EXPECT().RoundTrip(matcher).Return(&http.Response{
```

Comment:
Might it be worth making this call return a 404 such that we can then check that the next call for the other URL uses the other token?

Reply:
@DolceTriade ^

Reply:
Good idea. I took a stab at testing the fallback functionality. WDTY?

Reply:
Looks good to me, my only minor comment is that I don't think the `Body` and `ContentLength` are needed on the failing call, but it doesn't break anything so it's up to you if you take that out or not.

Reply:
Removed.